### PR TITLE
Globalization enhancements

### PR DESCRIPTION
**Pull request by francoislehoux-okta**
### Description
- `OdysseyTranslationProvider` and `OdysseyProvider` can take `translationOverrides` prop in to optionally override key/value pairs for any language that is supported as follow:

```jsx
const translationOverrides = {
  en: {
    "fieldlabel.optional.text": "Not Required"
  }
}

// ...

<OdysseyTranslationProvider translationOverrides={translationOverrides}>
  // ...
</OdysseyTranslationProvider>
```

- Removed `fs-extra`, everything could be reimplemented with `fs` directly.
- Enhanced documentation for OdysseyTranslationProvider

### CODE REVIEW

File: packages/odyssey-react-mui/src/OdysseyTranslationProvider.tsx [hunk 1]
```diff
@@ -10,24 +10,53 @@
...
+  const bundle = resources[languageCode];
+  const overrides = translationOverrides[languageCode];
+  return {
+    ...bundle,
+    ...overrides,
+  };
+};
+
+export const OdysseyTranslationProvider: FC<OdysseyTranslationProviderProps> =
+  ({ children, languageCode, translationOverrides }) => {
```
Comment (KevinGhadyani-Okta) [inline packages/odyssey-react-mui/src/OdysseyTranslationProvider.tsx:43]: I've heard a lot of things from TS devs about not using `FC` for React components.

So I'd prefer this instead:
```suggestion
export const OdysseyTranslationProvider =
  ({ children, languageCode, translationOverrides }: OdysseyTranslationProviderProps) => {
```

File: packages/odyssey-react-mui/src/OdysseyTranslationProvider.tsx [hunk 2]
```diff
@@ -10,24 +10,53 @@
...
   languageCode?: SupportedLanguages;
-}) {
-  useEffect(() => {
-    // Defaults to the browser's language if available otherwise `en` will be used
-    i18n.changeLanguage(languageCode || window.navigator.language);
-  }, [languageCode]);
-
-  return <I18nextProvider i18n={i18n}>{children}</I18nextProvider>;
+  translationOverrides?: TranslationOverrides;
 }
```
Comment (KevinGhadyani-Okta) [inline packages/odyssey-react-mui/src/OdysseyTranslationProvider.tsx:28]: Can we make this a `type` and not an `interface`?

File: packages/odyssey-react-mui/src/OdysseyTranslationProvider.tsx [hunk 3]
```diff
@@ -10,24 +10,53 @@
...
 import { I18nextProvider } from "react-i18next";
 
-export function OdysseyTranslationProvider({
-  children,
-  languageCode,
-}: {
+export type TranslationOverrides = {
+  [key in SupportedLanguages]?: Partial<typeof resources["en"]>;
+};
+
```
Comment (KevinGhadyani-Okta) [inline packages/odyssey-react-mui/src/OdysseyTranslationProvider.tsx:23]: Great TS type for `TranslationOverrides`! 👍

File: packages/odyssey-react-mui/src/OdysseyTranslationProvider.tsx [hunk 4]
```diff
@@ -10,24 +10,53 @@
...
+    useEffect(() => {
+      if (translationOverrides) {
+        Object.keys(translationOverrides).forEach((lng) => {
+          const bundle = mergeBundleOverrides(
+            lng as SupportedLanguages,
+            translationOverrides
+          );
+          i18n.addResourceBundle(lng, defaultNS, bundle);
+        });
```
Comment (KevinGhadyani-Okta) [inline packages/odyssey-react-mui/src/OdysseyTranslationProvider.tsx:57]: I'd prefer we make this more verbose:

```suggestion
        Object.keys(translationOverrides).forEach((language) => {
          const bundle = mergeBundleOverrides(
            language as SupportedLanguages,
            translationOverrides
          );
          i18n.addResourceBundle(language, defaultNS, bundle);
        });
```

File: packages/odyssey-storybook/src/guidelines/OdysseyTranslationProvider.stories.mdx [hunk 5]
```diff
@@ -56,3 +56,34 @@ const YourAppRoot = ({ children }) => (
...
+
+The following keys can have their values overriden:
+
+- `fielderror.screenreader.text`
+- `fieldlabel.optional.text`
+- `severity.success`
+- `severity.warning`
+- `severity.info`
+- `severity.error`
+- `toast.close.text`
```
Comment (KevinGhadyani-Okta) [inline packages/odyssey-storybook/src/guidelines/OdysseyTranslationProvider.stories.mdx:89]: It's nice we document these, but can we generate this from the TS instead?

I think we can do a loop from the `.ts` files.

File: packages/odyssey-react-mui/src/OdysseyTranslationProvider.tsx (hunk 4 above)
Comment (francoislehoux-okta) [reply packages/odyssey-react-mui/src/OdysseyTranslationProvider.tsx:57]: Right, that is possible as we renamed the prop to `languageCode` ;)

File: packages/odyssey-storybook/src/guidelines/OdysseyTranslationProvider.stories.mdx (hunk 5 above)
Comment (KevinGhadyani-Okta) [reply packages/odyssey-storybook/src/guidelines/OdysseyTranslationProvider.stories.mdx:89]: `.mdx` files allow you to write JavaScript.

I'll set this to **NON-BLOCKING**. Just make a ticket for it.

**Review by KevinGhadyani-Okta — APPROVED** [review]

File: packages/odyssey-storybook/src/guidelines/OdysseyTranslationProvider.stories.mdx (hunk 5 above)
Comment (francoislehoux-okta) [reply packages/odyssey-storybook/src/guidelines/OdysseyTranslationProvider.stories.mdx:89]: Looks like you can only render "JSX" components

**Pull request merged**